src/js/program/shared-media-pool.ts

```diff
@@ -0,0 +1,20 @@
+import { MediaElementPoolInt } from 'program/media-element-pool';
+
+export default function SharedMediaPool(sharedElement: HTMLVideoElement, mediaPool: MediaElementPoolInt): MediaElementPoolInt {
```

Comment:
It make make more sense to convert to classes. It looks like `MediaElementPoolInt` is making compromises (`OptionalUndefVideo` for a few methods that match across "classes") to fit these different modules.

Reply:
We don't have to go as far as making these classes, but it's worth considering what is cleaner. I _think_ we have good test coverage for these modules.

Reply:
agree that a class would be cleaner, i'm wishy-washy on scope creep for these changes. in general i think my approach is the less change to the underlying code the better given we aren't really expecting to QA these changes -- it might add a lot of overhead with a little return but i'm not sold on an opinion here

Reply:
OK let's not refactor, but try to lock down the types using `as` to assign so return types aren't wishy-washy. That will hurt us when we type the code using this module.

Reply:
removed the `OptionalUndef` in favor of `as`